### DISCUSSION TITLE
GraphQL not working with new Version

### DISCUSSION THREAD

**Discussion by Lucas-Gruber** [directus/directus#12337] [Archive]
GraphQL Requests no longer working east version 9.7.0

### To Reproduce

Use a Apollo GraphQL Client or a API-Client like Hopscotch and trie to connect to the directus.link/graphql or try to get any collection.

### Errors Shown

Web Console Error: [Error] Fetch API cannot load https://directus.domain.com/graphql?access_token=xxxxxxxxxxxx due to access control checks.

### What version of Directus are you using?

9.7.1

**Comment by rijkvanzanten** [comment]
Make sure to configure CORS 👍🏻 

As per the release notes https://github.com/directus/directus/releases/tag/v9.7.0, the default value for the CORS configuration has been changed for security reasons.

**Reply by Lucas-Gruber** [reply]
Thank you this fixed it, and sorry that I created a Issue for that. But I have one Problem with this. I have a local Dev Environment and a Live-Site. How can I use multiple Domains / IP-Adresses in CORS_ORIGIN?

**Reply by MoltenCoffee** [reply]
> How can I use multiple Domains / IP-Adresses in CORS_ORIGIN?

I believe you can use an `array:` prefix with a comma-seperated list, as follows: 

```
array:sub1.domain.com,sub2.domain.com,differentdomain.com
```

I got this from `api/src/env.ts`, perhaps this is something we should document more clearly (unless there are more convenient ways by now)

**Reply by rijkvanzanten** [reply]
> > How can I use multiple Domains / IP-Adresses in CORS_ORIGIN?
> 
> I believe you can use an `array:` prefix with a comma-seperated list, as follows:
> 
> ```
> array:sub1.domain.com,sub2.domain.com,differentdomain.com
> ```
> 
> I got this from `api/src/env.ts`, perhaps this is something we should document more clearly (unless there are more convenient ways by now)

CSVs are supposed to be automatically treated as arrays, but weren't. https://github.com/directus/directus/pull/12330 resolved that 👍🏻

**Reply by rijkvanzanten** [reply]
> got this from api/src/env.ts, perhaps this is something we should document more clearly (unless there are more convenient ways by now)

It's at the top of the config docs currently 🙂 

https://docs.directus.io/configuration/config-options/#environment-syntax-prefix


<img width="828" alt="CleanShot 2022-03-24 at 12 54 26@2x" src="https://user-images.githubusercontent.com/9141017/159969338-dad57d97-0d16-42a6-bff1-8178c500a6ec.png">

<img width="880" alt="CleanShot 2022-03-24 at 12 55 11@2x" src="https://user-images.githubusercontent.com/9141017/159969357-5db490ec-7190-411d-8b74-037b1c50dd35.png">

**Reply by MoltenCoffee** [reply]
Amazing!

**Reply by localjo** [reply]
Thanks for the information @rijkvanzanten 

How can we set CORS headers for our Directus Cloud instance?

**Reply by rijkvanzanten** [reply]
@localjo The default ones should work for most frontends. If you're running into specific issues you can hit up support@directus.io and they'll get it setup to your hearts content 🙂
